### DISCUSSION TITLE
keydown subscribe event in ReactJS

### DISCUSSION THREAD

**Discussion by huypham1411** [tradingview/lightweight-charts#1572] [Q&A]
Hi, is there any way to subscribe to the key-down event in this library? For example, hold x key to trigger a specific function?

**Comment by SlicedSilver** [comment]
The library doesn't offer an API for keyboard events directly but you can simply use the native APIs provided by the browser.

For example, our A11y guide has a section for adding keyboard press handlers: https://tradingview.github.io/lightweight-charts/tutorials/a11y/keyboard#adding-event-listener-for-keyboard-actions

- https://developer.mozilla.org/en-US/docs/Web/API/Element/keydown_event

**Reply by huypham1411** [reply]
thanks for your reply, this is not what I really want since the subscribe click provides param to trigger the function, is there any way to get this param value? I'm trying to create a measure box like tradingview using canvas
![image](https://github.com/tradingview/lightweight-charts/assets/54835897/d0531133-1bbd-439e-9cbc-d4bcedb34bcb)
